src/app/data.service.ts

```diff
@@ -1,19 +1,25 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { User } from './models/user';
+import { Post } from './models/post';
 
 @Injectable({
   providedIn: 'root'
 })
 export class DataService {
 
+  private rootUrl = 'https://jsonplaceholder.typicode.com';
+
   constructor(private http: HttpClient) {
   }
 
-  getUsersData() {
-    return this.http.get('http://jsonplaceholder.typicode.com/users');
+  getUserData(userId: number = null): Observable<User> {
```

Comment:
Both of the ajax calls follow the same format.  With parameter, return one, without parameter return all.  By changing the function signature, we allow for an optional parameter to be passed in.  When the parameter is present, we use it and when it is is not it defaults to null and we ignore it in the url.